darwin/dataset/download_manager.py

```diff
@@ -209,3 +266,28 @@ def download_image(url: str, path: Path, api_key: str) -> None:
         if time.time() - start > TIMEOUT:
             raise Exception(f"Timeout url request ({url}) after {TIMEOUT} seconds.")
         time.sleep(1)
+
+
+def _download_image(url: str, path: Path, api_key: str) -> None:
+    if path.exists():
+        return
+    TIMEOUT: int = 60
+    start: float = time.time()
+    while True:
```

Comment:
Does this mean that it will attempt 50~60 requests if the response is always HTTP 500?

Reply:
Also, we should not have `requests.get` here, this HTTP request needs to go to `client.py`.
Good catch!
Any ideas for the new function?